src/services/dogs.js

```diff
@@ -0,0 +1,11 @@
+import { checkError, client } from './client';
+
+export async function fetchDogs() {
+  const response = await client.from('dogs').select().order('name');
+  return checkError(response);
+}
+
+export async function fetchDogById(id) {
+  const response = await client.from('dogs').select('*').match({ id });
```

Comment:
If you append a .single() to this, you don't have to dog[0] later. 
.match() provides an array, and .single() will take the first result. 
Supabase seems to recommend using a .limit(1) in the chain also, for contingency